File: java/Análise Assintótica/BuscaOrdena.java
public class BuscaOrdena {
	public static void main(String[] args) {
		int v[] = {5, 3, 1, 9, -2, 3};
		
		System.out.println("Pos: " + busca(v, 9));
		
		for(int x = 0; x<v.length; x++) {
			System.out.print(v[x] + "; ");
		}
		
		ordena(v);

		
		for(int x = 0; x<v.length; x++) {
			System.out.print(v[x] + "; ");
		}
		
		System.out.println("Pos: " + busca(v, 9));
	}
	
	public static void ordena(int v[]) {
		
		int temp;
		
		for(int x = 0; x<v.length; x++) {
			for(int y=0;y<x; y++) {
				if(v[x] < v[y]) {
					temp = v[x];
					v[x] = v[y];
					v[y] = temp;
				}
			}
		}
	}
	
	public static int busca(int v[], int valor) {
		for(int x = 0; x<v.length; x++) {
			
				if(v[x] == valor) {
					return x;
				}
		}
		return -1;
	}
}
 

File: c/Análise Assintótica/BuscaOrdena.c
#include <stdio.h>

#define TAM_MAX 5
 
void print(int vetor[TAM_MAX]){
   for(int i=0;i<TAM_MAX;i++){
      printf("%i - %i\n", i, vetor[i]);
   }
}

void print1(int *v){
   
   for(int i=0;i<TAM_MAX;i++){
      printf("%i - %i\n", i, v[i]);
   }
}
// TAM_MAX = 10   (x== 9)
void ordena(int vetor[TAM_MAX]){
int temp;
		
   for(int x = 0; x<TAM_MAX; x++) {
      for(int y=0;y<x; y++) {
         if(vetor[x] < vetor[y]) {
            temp = vetor[x];
            vetor[x] = vetor[y];
            vetor[y] = temp;
         }
      }
   }
}

int busca(int *vetor, int valor) {
   for(int x = 0; x<TAM_MAX; x++) {		
      if(vetor[x] == valor) {
      return x;
      }
   }
   return -1;
}

int main() {


   int vetor[] = {2, 6, 1, -9, 4};

   print1(vetor);
   printf("\nBusca -9: %i", busca(vetor, -9));
   printf("\n\nOrdenado: \n");
   ordena(vetor);
   print1(vetor);
   printf("\nBusca -9: %i\n\n", busca(vetor, -9));
   return 0;
}

